custom/icds_reports/utils/aggregation_helpers/agg_ls_data.py

```diff
@@ -0,0 +1,214 @@
+                FROM "{ls_awc_mgt_ucr}"
+                WHERE submitted_on > %(start_date)s AND  submitted_on< %(end_date)s
+                AND location_entered is not null and location_entered <> ''
+                GROUP BY location_id
+            ) ut
+            WHERE agg_ls_report.supervisor_id = ut.supervisor_id
+        """.format(
+            tablename=self.tablename,
+            ls_awc_mgt_ucr=self._ucr_tablename(ucr_id=self.ls_awc_mgt_ucr)
+        ), {
+            "start_date": self.month_start,
+            "end_date": self.next_month_start
+        }
+
+        yield """
+            UPDATE "{tablename}" agg_ls_report
+            SET beneficiary_vists = ut.beneficiary_vists
+            FROM (
+                SELECT
+                count(*) as beneficiary_vists,
+                location_id as supervisor_id
+                FROM "{ls_home_visit_ucr}"
+                WHERE submitted_on > %(start_date)s AND  submitted_on< %(end_date)s
+                AND visit_type_entered is not null AND visit_type_entered <> ''
+                GROUP BY location_id
+            ) ut
+            WHERE agg_ls_report.supervisor_id = ut.supervisor_id
+        """.format(
+            tablename=self.tablename,
+            ls_home_visit_ucr=self._ucr_tablename(ucr_id=self.ls_home_visit_ucr)
+        ), {
+            "start_date": self.month_start,
+            "end_date": self.next_month_start
+        }
+
+    def indexes(self, aggregation_level):
+        """
+        Returns queries to create indices  for columns
+        district_id, block_id, supervisor_id and state_id based on
+        aggregation level
+        """
+        indexes = []
+        agg_locations = ['state_id']
+        if aggregation_level > 1:
+            indexes.append('CREATE INDEX ON "{}" (district_id)'.format(self._tablename_func(aggregation_level)))
+            agg_locations.append('district_id')
+        if aggregation_level > 2:
+            indexes.append('CREATE INDEX ON "{}" (block_id)'.format(self._tablename_func(aggregation_level)))
+            agg_locations.append('block_id')
+        if aggregation_level > 3:
+            indexes.append('CREATE INDEX ON "{}" (supervisor_id)'.format(self._tablename_func(aggregation_level)))
+            agg_locations.append('supervisor_id')
+
+        indexes.append('CREATE INDEX ON "{}" ({})'.format(self._tablename_func(aggregation_level),
+                                                          ', '.join(agg_locations)))
+        return indexes
+
+    def rollup_query(self, agg_level):
+        """
+        Returns the roll up query to the agg_level passed as argument.
+        Roll up query is used to roll up the data from supervisor level
+        to block level to district level to state level
+        """
+        locations = ['state_id', 'district_id', 'block_id', 'supervisor_id']
+
+        for i in range(3, agg_level - 1, -1):
+            locations[i] = "'All'"
+
+        return """
```

Comment:
Is there a reason you chose to do it this way instead of following the convention of the of the other agg tables (python tuples)?

Reply:
Ya, Because this table does not have the variety of calculations for columns only sum is there. Using this method makes it a smaller function